docs/manage/CLI.md

```diff
@@ -190,6 +193,107 @@ The `logout` command ends an authenticated CLI session
+viam module upload --organization=<org name> --name <module-id> --platform <platform> --version <version> --module meta.json <packaged-module.tar.gz>
+```
+
+Examples:
+
+```sh {class="command-line" data-prompt="$"}
+# create a new module named 'my-module' in organization 'abc':
+viam module create --organization='abc' --name 'my-module'
+
+# update the existing 'my-module' module with new changes to meta.json:
+viam module update --organization='abc' --name 'my-module'
+
+# upload a new or updated custom module 'my-module' to the Viam Registry:
+viam module upload --organization='abc' --platform "darwin/amd64" --version "1.0.0-rc1" --module meta.json packaged-module.tar.gz
+```
+
+#### Command options
+
+|        command option     |       description      | positional arguments
+| ----------- | ----------- | ----------- |
+| `create`    | create a new custom module on your local filesystem  | - |
+| `update`    | update an existing custom module on your local filesystem with recent changes to <file>meta.json</file> | - |
+| `upload`    | upload a new or existing custom module on your local filesystem to the Viam Registry (see [Using the upload command](#using-the-upload-command)) |
+| `help`      | return help      | - |
+
+##### Named arguments
+
+|        argument     |       description | applicable commands | required
+| ----------- | ----------- | ----------- | ----------- |
+| `--organization`      | organization name that the robot belongs to       |`create`, `update`, `upload`|true |
+| `--name`     |  name of the custom module to be created, updated, or uploaded    |`create`, `update`|true |
+| `--platform`      |  the platform to encode the resulting module binary as   |`upload`|true |
+| `--version`      |  the version of your module to set for this upload  |`upload`|true |
+| `--module`      |  the payload of the module to upload, including the <file>meta.json</file> configuration file and the packaged module as a `tar.gz` file    |`upload`|true |
+
+See [Using the upload command](#using-the-upload-command) for more information on using these arguments with `module upload`.
+
+##### Custom module creation workflow example
+
+To upload a new module to the Viam Registry, you need to `create` it with a new name, `update` it with your customized changes, and `upload` it to the Viam Registry.
+The following example demonstrates this workflow for an example module `my-first-module`:
+
+```sh {class="command-line" data-prompt="$"}
+## Create a new module 'my-first-module' locally, which creates a placeholder `meta.json` file on the local filesystem:
+viam module create --name 'my-first-module' --org_id 'abc'
+
+## Edit the newly-created meta.json file with the custom module-specific configuration:
+vi meta.json
+
+## Update the module with the new configuration:
+viam module update --name 'my-first-module' --org_id 'abc'
+
+## Package the module as a tar.gz archive:
+tar -zcf my-first-module.tar.gz my-first-module
```

Comment:
Adding this in here as a guess. 

Q: Is the `meta.json` file _in_ the module directory (`my-first-module` here), or _outside_ it? (you can guess why, in the `tgz` step, I'm asking this!)